### PR TITLE
feat: display topics prominently

### LINKED ISSUE #201: Set max width for questions and descriptions for desktop
Set max px width or max text count length to optimise readability for desktop / tablet

### LINKED ISSUE #255: Add "Category" filter for mobile
Currently the "Topics" selection can only be seen on desktop

### LINKED ISSUE #488: Mobile layout should have tag selector between nav bar and question list
Self-explanatory - rather than hide the tag selector, place it in the layout between nav bar and question list. 

This encourages users who are browsing to narrow their search by topic, allowing them to more quickly get to answers

### LINKED ISSUE #489: Pages by tag: Replace text heading with tag dropdown
To allow users to quickly switch between tags, replace the text header `Questions related to: xxx` with a dropdown component that, on entry selection, navigates to the appropriate tag

### LINKED ISSUE #491: QuestionList: Remove tag badges from entries
Blocked by #489

### PR DESCRIPTION
**Pull request by LoneRifle**
## Problem

The tag selection menu was not showing up on mobile, making it difficult for users to select a topic. Also, the existing layout was cluttered (e.g. with tag badges, view counts)

closes #491 
closes #489 
closes #488 
closes #201
closes #255 

## Solution
- Revamp tag selector layout
- Introduce tag selector in mobile layout
- Remove tag badges from entries
- Remove view count from entries
- Add tag description
- Update UI according to [new design layout](https://www.figma.com/file/Y2sqYzAkssq5xSFTN9KZeX/AskGov-Design-Master-v1?node-id=5655%3A7449)
  - update positioning of components (`AgencyLogo`, `SearchBarComponent`)
  - update default dimensions of the questions list
  - change colour of topics in menu upon hover

Details...
- A separate TagMenu component was created as its behaviour is different from the TagPanel component on the landing page. This TagMenu component comprises a `ref` which collapses the accordion menu upon clicking a `tag`. This `ref` was necessary as the current [Chakra accordion](https://chakra-ui.com/docs/disclosure/accordion) does not take in a `props` which allows it to close upon command

## Screenshots
Mobile:

https://user-images.githubusercontent.com/56983748/136530531-3d9367e1-acfb-4d48-9601-526d302f8a95.mov

Desktop:

https://user-images.githubusercontent.com/56983748/136530515-4ba01f74-f559-4627-b6b3-96837c44f28c.mov

## Tests
- Navigate to the home page and tag-specific page, check styling against figma
- Tag menu (and description, if applicable) should show up before the list of questions 
- Clicking on a topic should bring the user to a new page and collapse the tag menu
- Clicking "View All Questions" brings the user to a new page with "Select A Topic" as the tag menu heading

### CODE REVIEW

**Comment by LinHuiqing** [comment]
Screenshot 2021-10-08 at 5 06 45 PM

@wanlingt the search bar centering got lost

**Comment by LoneRifle** [comment]
Works for me..
image

**Comment by LinHuiqing** [comment]
Oh I forgot to rebase after fetching :') sorry

File: client/src/pages/HomePage/HomePage.component.jsx [hunk 1]
```diff
@@ -86,79 +101,70 @@ const HomePage = ({ match }) => {
...
+        ) : null}
       </Box>
+      <Box flex="1">{hasTagsKey ? <TagMenu /> : <TagPanel />}</Box>
       <Flex
-        maxW="1504px"
+        maxW="680px"
         m="auto"
-        w="100vw"
+        w="100%"
         pt={agency ? { base: '44px', md: '104px', xl: '80px' } : '44px'}
```
Comment (LoneRifle) [inline client/src/pages/HomePage/HomePage.component.jsx]: I think we can just have 40px padding here.
image

**Comment by LinHuiqing** [comment]
Screenshot 2021-10-08 at 5 36 29 PM

"Questions on this topic" should be caps and with "subhead-3" styling. Also, do we need to migrate the db to include `description` for tags?

**Comment by LinHuiqing** [comment]
Screenshot 2021-10-08 at 5 40 45 PM

"Top questions" on this page should be "ALL QUESTIONS" with styling: https://www.figma.com/file/Y2sqYzAkssq5xSFTN9KZeX/AskGov-Design-Master-v1?node-id=5660%3A14157

**Comment by LinHuiqing** [comment]
(not as important)

Screenshot 2021-10-08 at 5 43 49 PM

1. Main mobile page search box spacing has poofed
2. Search box should not have a black outline

**Comment by LoneRifle** [comment]
> Also, do we need to migrate the db to include `description` for tags?

Nope, already in schema

> 1. Main mobile page search box spacing has poofed

Only occurs on main landing page, not agency-specific page

> 2. Search box should not have a black outline

the black outline is an unrelated issue, ignore for now

**Review by LinHuiqing — APPROVED** [review]
Lgtm 👍 Other than last comment about search box spacing on main AskGov page

**Comment by LoneRifle** [comment]
image
Applied agency padding for landing page too

**Pull request merged**